packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -74,123 +90,522 @@ const propTypes = {
+      onCalendarButtonClick(event, onClick);
+    }
+  };
+
+  const handleOnButtonKeyDown = (event) => {
+    if (onKeyDown) {
+      onKeyDown(event);
+    }
+  };
+
+  const dateInputContainerClasses = cx([
+    'date-input-container',
+    theme.className,
+  ]);
+
+  const dateInputClasses = cx([
+    'date-input',
+    { 'is-focused': isFocused },
+    { 'is-invalid': isInvalid },
+    { 'is-incomplete': isIncomplete && required && !isInvalid },
+  ]);
+
+  const buttonClasses = cx([
+    'button',
+    { 'is-invalid': isInvalid },
+  ]);
+
+  const dayInputClasses = cx([
+    'date-input-day',
+    { 'initial-focus': dayInitialFocused },
+  ]);
+
+  const monthInputClasses = cx([
+    'date-input-month',
+    { 'initial-focus': monthInitialFocused },
+  ]);
+
+  const yearInputClasses = cx([
+    'date-input-year',
+    { 'initial-focus': yearInitialFocused },
+  ]);
+
+  const dateDayInput = (
+    <Input
+      {...additionalInputProps}
+      refCallback={setDayRef}
+      className={dayInputClasses}
+      type="text"
+      name={`terra-date-day-${name}`}
+      value={date.day}
+      onChange={handleDayChange}
+      placeholder={placeholderValues.day}
+      onFocus={(e) => handleOnInputFocus(e, DateUtil.inputType.DAY)}
+      onBlur={(e) => handleOnInputBlur(e, DateUtil.inputType.DAY)}
+      onKeyDown={handleDayInputKeydown}
+      maxLength="2"
+      size="2"
+      pattern="\d*"
+      aria-label={intl.formatMessage({ id: 'Terra.datePicker.dayLabel' })}
+    />
+  );
+
+  const dateMonthInput = (
+    <Input
+      {...additionalInputProps}
+      refCallback={setMonthRef}
+      className={monthInputClasses}
+      type="text"
+      name={`terra-date-month-${name}`}
+      value={date.month}
+      onChange={handleMonthChange}
+      placeholder={placeholderValues.month}
+      onFocus={(e) => handleOnInputFocus(e, DateUtil.inputType.MONTH)}
+      onBlur={(e) => handleOnInputBlur(e, DateUtil.inputType.MONTH)}
+      onKeyDown={handleMonthInputKeydown}
+      maxLength="2"
+      size="2"
+      pattern="\d*"
+      aria-label={intl.formatMessage({ id: 'Terra.datePicker.monthLabel' })}
+    />
+  );
+
+  const dateYearInput = (
+    <Input
+      {...additionalInputProps}
+      refCallback={setYearRef}
+      className={yearInputClasses}
+      type="text"
+      name={`terra-date-year-${name}`}
+      value={date.year}
+      onChange={handleYearChange}
+      placeholder={placeholderValues.year}
+      onFocus={(e) => handleOnInputFocus(e, DateUtil.inputType.YEAR)}
+      onBlur={(e) => handleOnInputBlur(e, DateUtil.inputType.YEAR)}
+      onKeyDown={handleYearInputKeydown}
+      maxLength="4"
+      size="4"
+      pattern="\d*"
+      aria-label={intl.formatMessage({ id: 'Terra.datePicker.yearLabel' })}
+    />
+  );
+
+  const dateSpacer = <span className={cx('date-spacer')}>{dateSeparator}</span>;
+
+  const dateInputFormat = DateUtil.getInputLayout(
```

Comment:
Should this be a new component instead of a new method on the Utils? This would remove the need to add React to the utils & convert it to jsx.

Reply:
Even i was hesitant on making DateUtils a jsx file, but i am not sure about making it a new component.
@benbcai, your thoughts on this?

Reply:
Separated the logic to be a new sub-component and reverted DateUtils to be a js file, as seen here : https://github.com/cerner/terra-framework/pull/1208/commits/004bfaa279f9550825ecf95e11b020f29aae7d0e